python/sedonadb: Several tests fail after pandas 3.0 release

As reported in https://github.com/apache/sedona-db/pull/533#issuecomment-3782577086 !

https://github.com/apache/sedona-db/actions/runs/21236733252/job/61106200223?pr=533

```
sedonadb/python/sedonadb/dataframe.py:368: SedonaError
=========================== short test summary info ============================
FAILED sedonadb/tests/test_dataframe.py::test_dataframe_from_pandas - sedonadb._lib.SedonaError: SedonaDB internal error: Can't scan RecordBatchReader provider more than once.
This issue was likely caused by a bug in SedonaDB's code. Please help us to resolve this by filing a bug report in our issue tracker: https://github.com/apache/sedona-db/issues
FAILED sedonadb/tests/test_datasource.py::test_read_ogr_multi_file - sedonadb._lib.SedonaError: it is not yet supported to write to hive partitions with datatype LargeUtf8
```

--- a/python/sedonadb/python/sedonadb/dataframe.py
+++ b/python/sedonadb/python/sedonadb/dataframe.py
@@ -518,7 +518,10 @@ def _qualified_type_name(obj):
 
 SPECIAL_CASED_SCANS = {
     "pyarrow.lib.Table": _scan_collected_default,
+    # pandas < 3.0
     "pandas.core.frame.DataFrame": _scan_collected_default,
+    # pandas >= 3.0
+    "pandas.DataFrame": _scan_collected_default,
     "geopandas.geodataframe.GeoDataFrame": _scan_geopandas,
     "polars.dataframe.frame.DataFrame": _scan_collected_default,
 }

--- a/python/sedonadb/tests/test_datasource.py
+++ b/python/sedonadb/tests/test_datasource.py
@@ -74,8 +74,13 @@ def test_read_ogr_multi_file(con):
 
     with tempfile.TemporaryDirectory() as td:
         # Create partitioned files by writing Parquet first and translating
-        # one file at a time
-        con.create_data_frame(gdf).to_parquet(td, partition_by="partition")
+        # one file at a time. We need to cast partition in pandas>=3.0 because
+        # the default translation of a string column is LargeUtf8 and this is not
+        # currently supported by DataFusion partition_by.
+        con.create_data_frame(gdf).to_view("tmp_gdf", overwrite=True)
+        con.sql(
+            """SELECT idx, partition::VARCHAR AS partition, wkb_geometry FROM tmp_gdf"""
+        ).to_parquet(td, partition_by="partition")
         for parquet_path in Path(td).rglob("*.parquet"):
             fgb_path = str(parquet_path).replace(".parquet", ".fgb")
             con.read_parquet(parquet_path).to_pandas().to_file(fgb_path)

